File: src/main/java/nl/hsleiden/model/Item.java
package nl.hsleiden.model;

public class Item {
    private int itemID;
    private String itemName;
    private String itemDescription;
    private double itemPrice;
    private String itemImage;
    private String itemType;
    private int itemAnimatorID;

    public Item(int itemID, String itemName, String itemDescription, double itemPrice, String itemImage, String itemType, int itemAnimatorID) {
        this.itemID = itemID;
        this.itemName = itemName;
        this.itemDescription = itemDescription;
        this.itemPrice = itemPrice;
        this.itemImage = itemImage;
        this.itemType = itemType;
        this.itemAnimatorID = itemAnimatorID;
    }

    public int getItemID() {
        return itemID;
    }

    public String getItemName() {
        return itemName;
    }

    public String getItemDescription() {
        return itemDescription;
    }

    public double getItemPrice() {
        return itemPrice;
    }

    public String getItemImage() {
        return itemImage;
    }

    public String getItemType() {
        return itemType;
    }

    public int getItemAnimatorID() {
        return itemAnimatorID;
    }
}

File: webshop/src/app/shared/modelsAndTheirServices/item.ts
export class Item {

  private itemID;
  private itemName;
  private itemDescription;
  private itemPrice;
  private itemImage;
  private itemType;
  private itemAnimatorID;

  // constructor(private itemID?: Number, private itemName?: String, private itemDescription?: String, private itemPrice?: Number, private itemImage?: String, private itemType?: String, private itemAnimatorID?: Number) {
  //
  // }

  constructor(private data?: object) {
    console.log(data);
    this.itemID = data.itemID;
    this.itemName = data.itemName;
    this.itemDescription = data.itemDescription;
    this.itemPrice = data.itemPrice;
    this.itemImage = data.itemImage;
    this.itemType = data.itemType;
    this.itemAnimatorID = data.itemAnimatorID;
  }

  public getData() {
    const data = {
      itemID: this.itemID,
      itemName: this.itemName,
      itemDescription: this.itemDescription,
      itemPrice: this.itemPrice,
      itemImage: this.itemImage,
      itemType: this.itemType,
      itemAnimatorID: this.itemAnimatorID
    };
    return data;
  }

  public getItemID() {
    return this.itemID;
  }

  public getItemName() {
    return this.itemName;
  }

  public getItemDescription() {
    return this.itemDescription;
  }

  public getItemPrice() {
    return this.itemPrice;
  }

  public getItemImage() {
    return this.itemImage;
  }

  public getItemType() {
    return this.itemType;
  }

  public getItemAnimatorId() {
    return this.itemAnimatorID;
  }
}
